src/hardware/setup.rs

```diff
@@ -992,9 +1009,33 @@ pub fn setup(
             output_voltage,
             output_current,
         );
+
+        let i2c_manager =
+            shared_bus_rtic::new!(i2c1, hal::i2c::I2c<hal::stm32::I2C1>);
+
+        let lm75 =
+            lm75::Lm75::new(i2c_manager.acquire(), lm75::Address::default());
+
+        let mcp23008 = mcp23017::MCP23017::new_default(
+            i2c_manager.acquire(),
+            mcp23017::Variant::MCP23008,
+        )
+        .unwrap();
+
+        let mcp_mutex =
```

Comment:
I don't think this needs to be a static singleton. Just make it, then move it into I2cDevices. You should still be able to create references to it when making `Relay::new()`.

Reply:
Even after some discussion with @ryan-summers I couldn't figure out how to get by without a singleton.

Reply:
Ack. Certainly fine as it is now.